### DISCUSSION TITLE
CollisionPair ordering?

### DISCUSSION THREAD

**Discussion by truher** [dyn4j/dyn4j#290] [Q&A] [answered]
Hi, I'm using a CollisionListener and Sensor to keep score in a simple game, similar to the basketball example.

It seems like that example (and other samples I've found on github) make some assumption about the identity of the "first" and "second" item in the collision data pair: only the first item is tested to see if it's the ball.

In my own experiments, it seems like the "first" item is often (always?) the faster-moving body, so that the fixed bodies in my simulation seem like they are always the second item.

I can't find anywhere in the dyn4j code where this invariant is enforced, and so I imagine the right thing to do in my CollisionListener is to look at both possible orderings of the items in the pair.  But I'm surprised that the examples don't do that, and that they seem to work anyway.

WDYT?

**Comment by wnbittle** [comment, marked as answer]
> But I'm surprised that the examples don't do that, and that they seem to work anyway.

I'm also surprised to find this...

> In my own experiments, it seems like the "first" item is often (always?) the faster-moving body, so that the fixed bodies in my simulation seem like they are always the second item.

I only looked at the `BasketBall` sample, but that one "works" because of the way the collision detection builds the pairs.  Every time step the `BroadphaseCollisionDetector` (in the default case `DynamicAABBTree`) will start the detection with only the bodies that _moved_.  Since this particular sample only has _one_ moving body (the basketballs) it's guaranteed that after the first time step the basketballs will always be Body1.

> I can't find anywhere in the dyn4j code where this invariant is enforced, and so I imagine the right thing to do in my CollisionListener is to look at both possible orderings of the items in the pair.

That said, you are 100% correct - this isn't something that should be relied on and I'll make sure to update all the samples to look at Body 1 and 2.

**Reply by truher** [reply]
perfect, thanks!
